### PR TITLE
Allow to specify additional parameters

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we allow users to specify additional parameters for
`elasticsearch.yml` and `jvm.options` in a generic fashion. We also add
the relevant documentation and a usage example so it is easier to get
started.

### CODE REVIEW

**Comment by balajismaniam** [comment]
+1

**Comment by danielmitterdorfer** [comment]
Thanks for the review. To your concern. I am aware of this but this should not pose a problem in practice as this is checked upon startup. If we set `discovery.zen.minimum_master_nodes: 1` twice, we get:

```
Failed to load settings from [elasticsearch.yml]]; nested: JsonParseException[Duplicate field 'discovery.zen.minimum_master_nodes'
 at [Source: sun.nio.ch.ChannelInputStream@37f21974; line: 73, column: 35]];
	at org.elasticsearch.common.settings.Settings$Builder.loadFromStream(Settings.java:1192)
[...]
Caused by: com.fasterxml.jackson.core.JsonParseException: Duplicate field 'discovery.zen.minimum_master_nodes'
```

File: cars/v1/vanilla/templates/config/elasticsearch.yml [hunk 1]
```diff
@@ -101,6 +101,14 @@ discovery.zen.minimum_master_nodes: {{ minimum_master_nodes }}
...
 {%- for key, value in cluster_settings.items() %}
 {{key}}: {{value | tojson}}
 {%- endfor %}
+
+{# cluster settings provided by the user via the command line #}
+{%- if additional_cluster_settings is defined %}
+{%- for key, value in additional_cluster_settings.items() %}
+{{key}}: {{value | tojson}}
+{%- endfor %}
+{%- endif %}
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply cars/v1/vanilla/templates/config/elasticsearch.yml]: Darn, those newlines...

**Comment by danielmitterdorfer** [comment]
The changes have been backported to all branches. As Elasticsearch 1.x and 2.x did not have a `jvm.options` file and this is a rather special use case, it will not be possible to override JVM options for 1.x. and 2.x, only cluster settings. For all versions starting with 5.0.0 both, cluster settings and JVM options, can be overridden.

**Pull request merged**